The job **`x86_64-gnu-llvm-15`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/6377972877/job/17307654726) [(plain)](https://github.com/rust-lang/rust/commit/6c95af33f93b78c236d437920ac39814ef36178a/checks/17307654726/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [assembly] tests/assembly/stack-protector/stack-protector-heuristics-effect.rs#strong stdout ----

error in revision `strong`: verification with 'FileCheck' failed
status: exit status: 1
command: "/usr/lib/llvm-15/bin/FileCheck" "--input-file" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/stack-protector/stack-protector-heuristics-effect.strong/stack-protector-heuristics-effect.s" "/checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs" "--allow-unused-prefixes" "--check-prefixes" "CHECK,NONMSVC,strong" "--dump-input-context" "100"
--- stderr -------------------------------
/checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs:375:17: error: CHECK-NOT: excluded string found in input
 // strong-NOT: __stack_chk_fail
                ^
                ^
/checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/stack-protector/stack-protector-heuristics-effect.strong/stack-protector-heuristics-effect.s:583:8: note: found here
 callq __stack_chk_fail@PLT

Input file: /checkout/obj/build/x86_64-unknown-linux-gnu/test/assembly/stack-protector/stack-protector-heuristics-effect.strong/stack-protector-heuristics-effect.s
Check file: /checkout/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs


-dump-input=help explains the following input dump.
Input was:
<<<<<<
         .
         .
         .
         .
       483:  popq %rbx 
       484:  .cfi_def_cfa_offset 8 
       485:  jmpq *%rax 
       486: .Lfunc_end16: 
       487:  .size alloca_dynamic_arg, .Lfunc_end16-alloca_dynamic_arg 
       488:  .cfi_endproc 
       489:  
       490:  .section .text.unsized_fn_param,"ax",@progbits 
       491:  .globl unsized_fn_param 
       492:  .p2align 4, 0x90 
       493:  .type unsized_fn_param,@function 
       494: unsized_fn_param: 
       494: unsized_fn_param: 
       495: .Lfunc_begin1: 
       496:  .cfi_startproc 
       497:  .cfi_personality 155, DW.ref.rust_eh_personality 
       498:  .cfi_lsda 27, .Lexception1 
       499:  pushq %r15 
       500:  .cfi_def_cfa_offset 16 
       501:  pushq %r14 
       502:  .cfi_def_cfa_offset 24 
       503:  pushq %r12 
       504:  .cfi_def_cfa_offset 32 
       505:  pushq %rbx 
       506:  .cfi_def_cfa_offset 40 
       507:  subq $24, %rsp 
       508:  .cfi_def_cfa_offset 64 
       509:  .cfi_offset %rbx, -40 
       510:  .cfi_offset %r12, -32 
       511:  .cfi_offset %r14, -24 
       512:  .cfi_offset %r15, -16 
       513:  movq %fs:40, %rax 
       514:  movq %rax, 16(%rsp) 
       515:  movl %edx, %eax 
       516:  movl $2, %r12d 
       517:  subq %rax, %r12 
       518:  cmpq %rsi, %r12 
       519:  ja .LBB17_8 
       520:  movq %rcx, %r14 
       521:  movq %rdi, %r15 
       522:  movq __rust_no_alloc_shim_is_unstable@GOTPCREL(%rip), %rax 
       523:  movzbl (%rax), %eax 
       524:  movl $1, %esi 
       525:  movq %r12, %rdi 
       526:  callq *__rust_alloc@GOTPCREL(%rip) 
       527:  testq %rax, %rax 
       528:  je .LBB17_2 
       529:  movq %rax, %rbx 
       530:  subq %r15, %rax 
       531:  negq %rax 
       532:  movq %rbx, %rcx 
       533:  subq %r15, %rcx 
       534:  cmovbeq %rax, %rcx 
       535:  cmpq %r12, %rcx 
       536:  jb .LBB17_3 
       537:  movq %rbx, %rdi 
       538:  movq %r15, %rsi 
       539:  movq %r12, %rdx 
       540:  callq *memcpy@GOTPCREL(%rip) 
       541:  movq %rbx, (%rsp) 
       542:  movq %r12, 8(%rsp) 
       543: .Ltmp3: 
       544:  movq %rbx, %rdi 
       545:  movq %r12, %rsi 
       546:  callq *%r14 
       547: .Ltmp4: 
       548:  movl $1, %edx 
       549:  movq %rbx, %rdi 
       550:  movq %r12, %rsi 
       551:  callq *__rust_dealloc@GOTPCREL(%rip) 
       552:  movq %fs:40, %rax 
       553:  cmpq 16(%rsp), %rax 
       554:  jne .LBB17_7 
       555:  addq $24, %rsp 
       556:  .cfi_def_cfa_offset 40 
       557:  popq %rbx 
       558:  .cfi_def_cfa_offset 32 
       559:  popq %r12 
       560:  .cfi_def_cfa_offset 24 
       561:  popq %r14 
       562:  .cfi_def_cfa_offset 16 
       563:  popq %r15 
       564:  .cfi_def_cfa_offset 8 
       565:  retq 
       566: .LBB17_8: 
       567:  .cfi_def_cfa_offset 64 
       568:  leaq .L__unnamed_1(%rip), %rdx 
       569:  movq %r12, %rdi 
       570:  callq *_ZN4core5slice5index24slice_end_index_len_fail17h50ad446730a31f8bE@GOTPCREL(%rip) 
       571:  ud2 
       572: .LBB17_2: 
       573:  movl $1, %edi 
       574:  movq %r12, %rsi 
       575:  callq *_ZN5alloc5alloc18handle_alloc_error17h3df0feeffcb7a2f5E@GOTPCREL(%rip) 
       576:  ud2 
       577: .LBB17_3: 
       578:  leaq .L__unnamed_2(%rip), %rdi 
       579:  movl $166, %esi 
       580:  callq *_ZN4core9panicking14panic_nounwind17ha05790965db3321fE@GOTPCREL(%rip) 
       581:  ud2 
       582: .LBB17_7: 
       583:  callq __stack_chk_fail@PLT 
not:375            !~~~~~~~~~~~~~~~      error: no match expected
       584:  ud2 
       585: .LBB17_9: 
       586: .Ltmp5: 
       587:  movq %rax, %rbx 
       588:  movq %rsp, %rdi 
       589:  callq _ZN72_$LT$alloc..boxed..Box$LT$T$C$A$GT$$u20$as$u20$core..ops..drop..Drop$GT$4drop17h1d89ed2346fed321E 
       590:  movq %rbx, %rdi 
       591:  callq _Unwind_Resume@PLT 
       592:  ud2 
       593: .Lfunc_end17: 
       594:  .size unsized_fn_param, .Lfunc_end17-unsized_fn_param 
       595:  .cfi_endproc 
       596:  .section .gcc_except_table.unsized_fn_param,"a",@progbits 
       597:  .p2align 2 
       599: .Lexception1: 
       600:  .byte 255 
       601:  .byte 255 
       602:  .byte 1 
       602:  .byte 1 
       603:  .uleb128 .Lcst_end1-.Lcst_begin1 
       604: .Lcst_begin1: 
       605:  .uleb128 .Lfunc_begin1-.Lfunc_begin1 
       606:  .uleb128 .Ltmp3-.Lfunc_begin1 
       607:  .byte 0 
       608:  .byte 0 
       609:  .uleb128 .Ltmp3-.Lfunc_begin1 
       610:  .uleb128 .Ltmp4-.Ltmp3 
       611:  .uleb128 .Ltmp5-.Lfunc_begin1 
       612:  .byte 0 
       613:  .uleb128 .Ltmp4-.Lfunc_begin1 
       614:  .uleb128 .Lfunc_end17-.Ltmp4 
       615:  .byte 0 
       616:  .byte 0 
       617: .Lcst_end1: 
       618:  .p2align 2 
       619:  
       620:  .section .text.unsized_local,"ax",@progbits 
       621:  .globl unsized_local 
       622:  .p2align 4, 0x90 
       623:  .type unsized_local,@function 
       625:  .cfi_startproc 
       625:  .cfi_startproc 
       626:  pushq %rbp 
       627:  .cfi_def_cfa_offset 16 
       628:  .cfi_offset %rbp, -16 
       629:  movq %rsp, %rbp 
       630:  .cfi_def_cfa_register %rbp 
       631:  pushq %r15 
       632:  pushq %r14 
       633:  pushq %r13 
       634:  pushq %r12 
       635:  pushq %rbx 
       636:  pushq %rax 
       637:  .cfi_offset %rbx, -56 
       638:  .cfi_offset %r12, -48 
       639:  .cfi_offset %r13, -40 
       640:  .cfi_offset %r14, -32 
       641:  .cfi_offset %r15, -24 
       642:  movq %fs:40, %rax 
       643:  movq %rax, -48(%rbp) 
       644:  movl %edx, %eax 
       645:  movl $2, %ebx 
       646:  subq %rax, %rbx 
       647:  cmpq %rsi, %rbx 
       648:  ja .LBB18_7 
       649:  movq %rcx, %r14 
       650:  movq %rdi, %r15 
       651:  movq __rust_no_alloc_shim_is_unstable@GOTPCREL(%rip), %rax 
       652:  movzbl (%rax), %eax 
       653:  movl $1, %esi 
       654:  movq %rbx, %rdi 
       655:  callq *__rust_alloc@GOTPCREL(%rip) 
       656:  testq %rax, %rax 
       657:  je .LBB18_2 
       658:  movq %rax, %r12 
       659:  subq %r15, %rax 
       660:  negq %rax 
       661:  movq %r12, %rcx 
       662:  subq %r15, %rcx 
       663:  cmovbeq %rax, %rcx 
       664:  cmpq %rbx, %rcx 
       665:  jb .LBB18_3 
       666:  leal 15(%rbx), %eax 
       667:  andl $-16, %eax 
       668:  callq __rust_probestack 
       669:  subq %rax, %rsp 
       670:  movq %rsp, %r13 
       671:  movq %r13, %rdi 
       672:  movq %r15, %rsi 
       673:  movq %rbx, %rdx 
       674:  callq *memcpy@GOTPCREL(%rip) 
       675:  movl $1, %edx 
       676:  movq %r12, %rdi 
       677:  movq %rbx, %rsi 
       678:  callq *__rust_dealloc@GOTPCREL(%rip) 
       679:  movq %r13, %rdi 
       680:  movq %rbx, %rsi 
       681:  callq *%r14 
       682:  movq %fs:40, %rax 
       683:  cmpq -48(%rbp), %rax 
         .
         .
>>>>>>
------------------------------------------
```

</details>

--- a/compiler/rustc_codegen_llvm/src/builder.rs
+++ b/compiler/rustc_codegen_llvm/src/builder.rs
@@ -3,6 +3,7 @@ use crate::attributes;
 use crate::common::Funclet;
 use crate::context::CodegenCx;
 use crate::llvm::{self, AtomicOrdering, AtomicRmwBinOp, BasicBlock, False, True};
+use crate::llvm_util;
 use crate::type_::Type;
 use crate::type_of::LayoutLlvmExt;
 use crate::value::Value;
@@ -1226,10 +1227,15 @@ impl<'a, 'll, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'll, 'tcx> {
     }
 
     fn apply_attrs_to_cleanup_callsite(&mut self, llret: &'ll Value) {
-        // Cleanup is always the cold path.
-        let cold_inline = llvm::AttributeKind::Cold.create_attr(self.llcx);
-
-        attributes::apply_to_callsite(llret, llvm::AttributePlace::Function, &[cold_inline]);
+        if llvm_util::get_version() < (17, 0, 2) {
+            // Work around https://github.com/llvm/llvm-project/issues/66984.
+            let noinline = llvm::AttributeKind::NoInline.create_attr(self.llcx);
+            attributes::apply_to_callsite(llret, llvm::AttributePlace::Function, &[noinline]);
+        } else {
+            // Cleanup is always the cold path.
+            let cold_inline = llvm::AttributeKind::Cold.create_attr(self.llcx);
+            attributes::apply_to_callsite(llret, llvm::AttributePlace::Function, &[cold_inline]);
+        }
     }
 }
 

--- a/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs
+++ b/tests/assembly/stack-protector/stack-protector-heuristics-effect.rs
@@ -9,6 +9,7 @@
 // [basic] compile-flags: -Z stack-protector=basic
 // [none] compile-flags: -Z stack-protector=none
 // compile-flags: -C opt-level=2 -Z merge-functions=disabled
+// min-llvm-version: 17.0.2
 
 #![crate_type = "lib"]
 

--- a/tests/codegen/issue-97217.rs
+++ b/tests/codegen/issue-97217.rs
@@ -1,5 +1,6 @@
 // compile-flags: -C opt-level=3
 // ignore-debug: the debug assertions get in the way
+// min-llvm-version: 17.0.2
 #![crate_type = "lib"]
 
 // Regression test for issue 97217 (the following should result in no allocations)

--- a/tests/codegen/unwind-landingpad-cold.rs
+++ b/tests/codegen/unwind-landingpad-cold.rs
@@ -1,5 +1,6 @@
 // compile-flags: -Cno-prepopulate-passes
 // needs-unwind
+// min-llvm-version: 17.0.2
 #![crate_type = "lib"]
 
 // This test checks that drop calls in unwind landing pads

--- a/tests/codegen/unwind-landingpad-inline.rs
+++ b/tests/codegen/unwind-landingpad-inline.rs
@@ -1,4 +1,4 @@
-// min-llvm-version: 15.0.0
+// min-llvm-version: 17.0.2
 // compile-flags: -Copt-level=3
 // ignore-debug: the debug assertions get in the way
 #![crate_type = "lib"]

